fmf/base.py

```diff
@@ -172,10 +172,29 @@ def _initialize(self, path):
 
     def _merge_plus(self, data, key, value, prepend=False):
         """ Handle extending attributes using the '+' suffix """
+
         # Nothing to do if key not in parent
         if key not in data:
             data[key] = value
             return
+
+        # Special handling for merging lists with dictionary
+        if isinstance(data[key], list) and isinstance(value, dict):
+            for list_item in data[key]:
+                if not isinstance(list_item, dict):
+                    "MergeError: Item '{0}' in {1} must be a dictionary.".format(
+                        list_item, self.name)
+                self._merge_special(list_item, value)
+            return
+        if isinstance(data[key], dict) and isinstance(value, list):
+            for list_item in value:
+                if not isinstance(list_item, dict):
+                    "MergeError: Item '{0}' in {1} must be a dictionary.".format(
+                        list_item, self.name)
+                self._merge_special(list_item, data[key])
```

Comment:
Just wondering, is this safe? I am modifying item from the list I am iterating through. But I am not adding any new items to the list so I hope this is fine. Or should I do a deepcopy instead?

Reply:
You should be safe as long as you are not changing the list itself, i.e. adding to it, removing from it, or replacing items with other objects. Changing an item itself, if it's a mutable value, should not cause any trouble.

Reply:
So I guess we should be fine.